### PR TITLE
Fix obsolete uses of cast and vector

### PR DESCRIPTION
**Pull request by r-owen**
These changes fall into 3 categories:
- Fix examples (or in the case of one test, fix what I can and file a ticket to fix the rest)
- Remove obsolete code that was commented out
- Remove a skipped test that will pybind11 compatible.

### CODE REVIEW

File: examples/estimateBackground.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
 
 def getImage():
     imagePath = os.path.join(lsst.utils.getPackageDir("afwdata"),
-                             "DC3a-Sim", "sci", "v5-e0", "v5-e0-c011-a00.sci_img.fits")
+                             "DC3a-Sim", "sci", "v5-e0", "v5-e0-c011-a00.sci.fits")
```
Comment (kfindeisen) [inline examples/estimateBackground.py]: Why is this change needed? It doesn't seem to be related either to casting or to Swig container wrappers...

File: examples/spatialCellExample.py [hunk 2]
```diff
@@ -62,13 +62,13 @@ def readImage(filename=None):
         except Exception:
             raise RuntimeError("You must provide a filename or setup afwdata to run these examples")
 
-        filename = os.path.join(afwDataDir, "CFHT", "D4", "cal-53535-i-797722_1")
+        filename = os.path.join(afwDataDir, "CFHT", "D4", "cal-53535-i-797722_1.fits")
```
Comment (kfindeisen) [inline examples/spatialCellExample.py]: Again, how is this pybind11-related?

File: examples/spatialCellExample.py [hunk 3]
```diff
@@ -62,13 +62,13 @@ def readImage(filename=None):
...
 
         bbox = afwGeom.Box2I(afwGeom.Point2I(270, 2530), afwGeom.Extent2I(512, 512))
     else:
         bbox = None
 
-    mi = afwImage.MaskedImageF(filename, 0, None, bbox, afwImage.LOCAL)
+    mi = afwImage.MaskedImageF(filename, bbox=bbox, origin=afwImage.LOCAL)
```
Comment (kfindeisen) [inline examples/spatialCellExample.py]: Nice use of keywords. :+1:

File: examples/estimateBackground.py (hunk 1 above)
Comment (r-owen) [inline examples/estimateBackground.py]: This change is needed because the file that was being loaded disappeared years ago (we used to persist MaskedImage and Exposure as three separate FITS files with suffixes `_img.fits, `_var.fits` and `_mask.fits`.

I will make this change (and the other one like it) as a separate commit. That should clear things up.
